app/src/main/java/com/zeronfinity/cpfy/MainActivity.kt

```diff
@@ -1,11 +1,78 @@
 package com.zeronfinity.cpfy
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.net.HttpURLConnection
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import kotlin.coroutines.CoroutineContext
+
+class MainActivity: AppCompatActivity(), CoroutineScope {
+    lateinit var job: Job
+
+    override val coroutineContext: CoroutineContext
+        get() = job + Dispatchers.Main
+
+    val contestApiUrl = "https://clist.by/api/v1/json/contest/"
+    val apiUsername = "Zeronfinity"
```

Comment:
I don't understand much about what you are trying to achieve here, but shouldn't it be dynamic?

Reply:
apiUsername?

Could you please suggest what you have in mind?